File: src/main/java/com/murex/tbw/domain/country/Country.java
package com.murex.tbw.domain.country;

public final class Country {
    private final String name;
    private final Currency currency;
    private final Language language;

    public Country(String name, Currency currency, Language language) {
        this.name = name;
        this.currency = currency;
        this.language = language;
    }

    public String getName() {
        return name;
    }

    public Language getLanguage() {
        return language;
    }

    public Currency getCurrency() {
        return currency;
    }
}

File: scala/src/main/scala/com/murex/tbw/domain/country/Country.scala
/*
 * ******************************************************************************
 *  *
 *  * Copyright (c) {2003-2022} Murex S.A.S. and its affiliates.
 *  * All rights reserved. This program and the accompanying materials
 *  * are made available under the terms of the MIT License
 *  * which accompanies this distribution, and is available at
 *  * https://opensource.org/licenses/MIT
 *  *
 *  ******************************************************************************
 */

package com.murex.tbw.domain.country

import com.murex.tbw.domain.country.Currencies.Currency
import com.murex.tbw.domain.country.Languages.Language

case class Country(name: String, currency: Currency, language: Language)
